### ISSUE TITLE
[Bug] Crash when loading up any file

### ISSUE THREAD

**Issue by pakoito** [WerWolv/ImHex#881]
### Operating System

Windows

### What's the issue you encountered?

New install using chocolatey, they program starts correctly and crash when a file, any file, is loaded from the start screen.

### How can the issue be reproduced?

Install using `choco install ImHex`
Open program
Close "Auto-Update" popup
Click on Load file
Program closes

### ImHex Version

1.26.2

### ImHex Build Type

- [ ] Nightly or built from sources

### Installation type

choco install ImHex

### Additional context?

I tried from a folder mounted as a drive in Windows and thought that was the cause, but it also repros for any other path.

**Comment by iTrooz** [comment]
Hey, could you try with the msi or portable package ? https://github.com/WerWolv/ImHex/releases

**Comment by pakoito** [comment]
Could NOT repro the crash on any of the releases above. It's a chocolatey problem, which is suprising

```
By installing, you accept licenses for the packages.
Progress: Downloading imhex 1.26.2... 100%

imhex v1.26.2 [Approved]
imhex package files install completed. Performing other installation steps.
Downloading imhex
  from 'https://github.com/WerWolv/ImHex/releases/download/v1.26.2/imhex-1.26.2-Windows-Portable.zip'
Progress: 100% - Completed download of C:\Users\paco\AppData\Local\Temp\chocolatey\imhex\1.26.2\imhex-1.26.2-Windows-Portable.zip (21.85 MB).
Download of imhex-1.26.2-Windows-Portable.zip (21.85 MB) completed.
Hashes match.
Extracting C:\Users\paco\AppData\Local\Temp\chocolatey\imhex\1.26.2\imhex-1.26.2-Windows-Portable.zip to C:\ProgramData\chocolatey\lib\imhex\tools...
C:\ProgramData\chocolatey\lib\imhex\tools
 ShimGen has successfully created a gui shim for imhex.exe
 The install of imhex was successful.
  Software installed to 'C:\ProgramData\chocolatey\lib\imhex\tools'

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
```

**Comment by WerWolv** [comment]
Interestingly enough, I can also not reproduce it with chocolatey. Just installed it on my machine, followed your instructions and everything seems to work like it should

**Comment by pakoito** [comment]
I did a second attempt with the check before. I'll have something wrong in the enviro. In any case the portable ones do the job, so thanks :D

**Comment by mi-hol** [comment]
> portable package NoGPU

fails too see #886 

tried with older versions i.e. 1.24.3.0 
Windows Eventlog has the following:
Fault bucket , type 0
Event Name: PCA2
Response: Not available
Cab Id: 0

Problem signature:
P1: imhex.exe
P2: 1.24.3.0
P3: ImHex Hex Editor
P4: ImHex
P5: WerWolv
P6: 100
P7: 0
P8: 
P9: 
P10:

**Comment by mi-hol** [comment]
> tried with older versions i.e. 1.24.3.0

reply from closed issue #886 was 
"Tried with older Windows versions but none of them even opens after clicking on exe in Explorer.

winget installs the GPU accelerated version of ImHex so if your system doesn't have a GPU, it won't run."

But in this wouldn't an error message indicating something like "no GPU available" display to give the user proper feedback?

**Comment by WerWolv** [comment]
Of course yeah. I'm trying to find out the thing that triggers that issue right now since I can't reproduce it on any machine I've tried it on so far

**Comment by mi-hol** [comment]
> I can't reproduce it on any machine I've tried it on so far

Thats is a common issue, still needing a code change in order to get more information required for troubleshooting

**Comment by mi-hol** [comment]
From my experience (operating large data centers) the top root cause for any outage, was "missed return code checking" leading to unexpected code behavior with no useful error message.

**Comment by WerWolv** [comment]
I'd say the error checking at the place where I assume the error ocurres is pretty okay.
![image](https://user-images.githubusercontent.com/10835354/212691618-57f478ef-5ab8-46b9-abde-ea24f58b1e02.png)

Could you please tell me now if it works when your Anti Virus tool is turned off?

**Comment by mi-hol** [comment]
> at the place where I assume the error ocurres is pretty okay

would "unhandled exception" not contradict the assumption?

**Comment by WerWolv** [comment]
There's nothing throwing an exception there. But an issue probably there _somewhere_ causes an exception to be thrown later on.

Please for fucking christs sake can you answer my question now? I can't debug any of this if you're not cooperating

**Comment by clagah** [comment]
> Could you please tell me now if it works when your Anti Virus tool is turned off?

I'm on a corporate computer and unable to disable AV, tried temporary to kill the processes but did not see any difference in behavior.

> Do you maybe have a Windows Defender or some other Anti Virus tool running in the background?

Of course, there is the risk of AV locking the file somehow, but why only when runnning from the version installed via choco and not when running the portable version?

**Comment by WerWolv** [comment]
Thank you. I don't know much about choco but it might be because ImHex gets installed in a different location in that case?
I'm not the one maintaining the choco package so that might rather be a question for them.

**Comment by iTrooz** [comment]
The other guy had a problem with the portable version in a system directory

Maybe this happens when ImHex is installed in a normally not-writtable location ?

**Comment by mi-hol** [comment]
I extracted Portable NoGPU version in "Download" folder and disabled Defender "RealTime Protection" and was able to open a file successfully.
As soon as I enable  Defender "RealTime Protection" get below

![image](https://user-images.githubusercontent.com/22799428/212696867-6bc54013-ab62-41b6-837a-6f4b2b3bee6d.png)

**Comment by mi-hol** [comment]
Further testing with disabled "RealTime Protection"

After I select a char in file App seems to loop indefinitly (consumes one core) and Windows title shows "not Responding"
![image](https://user-images.githubusercontent.com/22799428/212699046-adc69d07-aad0-43e0-abfd-4baf917a4cec.png)

**Comment by mi-hol** [comment]
> The ImHex log states:

@clagah How did you manage to get log?
Even after opening a file successfully `dir Downloads\imhex-1.26.2-Windows-Portable-NoGPU\logs` shows no log file :(

**Comment by clagah** [comment]
@mi-hol It was available under C:\Users\<username>\AppData\Local\ImHex\logs. I did not do anything special.

**Comment by WerWolv** [comment]
Okay if you guys want to try again, the latest nightly now properly outputs log information when an exception is thrown. Hopefully that will help identifying the underlying issue

**Comment by mi-hol** [comment]
would be happy to try but from clicking link in Install.md don't know where to find the zip?
![image](https://user-images.githubusercontent.com/22799428/212758573-5ac9a2a2-a291-4536-a9cb-68dd1f0607b6.png)

**Comment by mi-hol** [comment]
seems I found it and now it runs!

**Comment by mi-hol** [comment]
![image](https://user-images.githubusercontent.com/22799428/212761243-04126be0-5e7a-45dc-a6b5-9c45f8cba8e7.png)

log:
[21:16:42] [INFO]  [main] Welcome to ImHex 1.26.2!
[21:16:42] [INFO]  [main] Native scaling set to: 1.0
[21:16:42] [INFO]  [main] Task 'Setting up environment' finished in 1 ms
[21:16:42] [INFO]  [main] Task 'Creating directories' finished in 35 ms
[21:16:42] [INFO]  [main] Task 'Loading settings' finished in 1 ms
[21:16:42] [ERROR] [libimhex] Theme 'Dark' does not exist and no other themes are available!
[21:16:42] [INFO]  [main] Task 'Loading plugins' finished in 307 ms
[21:16:42] [INFO]  [main] Task 'Checking for updates' finished in 0 ms
[21:16:45] [INFO]  [main] Task 'Loading fonts' finished in 2924 ms

Would suggest to release this a new patch version  1.26.3

**Comment by clagah** [comment]
I was a bit late to the game yesterday and was not able to download the nightly with only this fix. it also included a fix related to where the logs are saved so I could not get anything useful out of it.

I tried to reproduce on another computer as well. Same problem with the choco installed version but no issue with the portable.

However, I moved the portable version in to a directory under C:\ProgramData where the choco installed version is also located. This directory structure is restricted so a regular user does not have Modify and Write security rights in this location. When running imhex from this location the problem appears again with the portable download that was working when running from another directory such as C:\Temp\.

However, as I mentioned, I did not get any logs at all with this nightly, probably because they were trying to be written at the location of the install, but the directories and files there are restricted.

So it appears that there is something requiring write or modify access to the binaries or files in the install directory that is causing the issue.

Hope that helps pin pointing what could be the root cause.

**Comment by WerWolv** [comment]
Thank you. To get log output, you can also just open a command prompt and run the ImHex executable from there. All the logs will then be dumped into the console

**Comment by clagah** [comment]
Yes, I found the previous build available as well now. Log contains:

> [08:44:14] [FATAL] [main] Program terminated with uncaught exception: std::bad_alloc()::what() -> std::bad_alloc
> [08:44:14] [FATAL] [main] Terminating with signal 'SIGABRT' (22)

**Comment by WerWolv** [comment]
Welp that is by far the most useless exception that could have been thrown :D
Anyway thank you for your time, I'll try to reproduce the issue myself again

**Comment by WerWolv** [comment]
Finally managed to reproduce it by creating a secondary, unelevated account. The latest commit should hopefully fix it

**Comment by mi-hol** [comment]
> unelevated account

and Real-Time protection is the standard Windows setup these day. 
Great this was found and fixed!
Hope the test environment will be adapted

**Comment by ZeChArtiahSaher** [comment]
same thing happening on osx
